core/src/main/scala/cats/data/Ior.scala

```diff
@@ -793,11 +793,12 @@ sealed abstract class Ior[+A, +B] extends Product with Serializable {
     }
 
   final def ===[AA >: A, BB >: B](that: AA Ior BB)(implicit AA: Eq[AA], BB: Eq[BB]): Boolean =
-    fold(
-      a => that.fold(a2 => AA.eqv(a, a2), b2 => false, (a2, b2) => false),
-      b => that.fold(a2 => false, b2 => BB.eqv(b, b2), (a2, b2) => false),
-      (a, b) => that.fold(a2 => false, b2 => false, (a2, b2) => AA.eqv(a, a2) && BB.eqv(b, b2))
-    )
+    (this, that) match {
+      case (Ior.Left(a), Ior.Left(aa))        => AA.eqv(a, aa)
+      case (Ior.Right(b), Ior.Right(bb))      => BB.eqv(b, bb)
+      case (Ior.Both(a, b), Ior.Both(aa, bb)) => AA.eqv(a, aa) && BB.eqv(b, bb)
+      case _                                  => false
```

Comment:
If we are going to discourage using `case _` then should I change it to `case (_, _) =>` or something?

Reply:
uh oh, here we have 3! cases to match. probably ф wildcard case is a trade-off here. `(_, _)` is fine with me.

Reply:
I think the wildcard should be fine here – since we construct the tuple in-place then it won't cause any additional value uncertainty like null values or something...